### ISSUE TITLE
Generalize V-REP C++ Robot definitions and plugins via RBDyn

### ISSUE THREAD

**Issue by ahundt** [ahundt/grl#114]
Robot definitions can automatically be extracted and built up using RBDyn as the API.

Key V-REP functions:

- [simGetObjectsInTree](http://www.coppeliarobotics.com/helpFiles/en/regularApi/simGetObjectsInTree.htm)
- [simGetObjectChild](http://www.coppeliarobotics.com/helpFiles/en/regularApi/simGetObjectChild.htm)
- [simGetObjectParent](http://www.coppeliarobotics.com/helpFiles/en/regularApi/simGetObjectParent.htm)
- [simGetObjectType](http://www.coppeliarobotics.com/helpFiles/en/regularApi/simGetObjectType.htm)
- [simGetJointType](http://www.coppeliarobotics.com/helpFiles/en/regularApi/simGetJointType.htm)

Key RBDyn Object:

- [MultiBodyGraph](https://github.com/jrl-umi3218/RBDyn/blob/master/src/MultiBodyGraph.h)

Basically, traverse the tree, and generate a MultiBodyGraph, then interact via that object. The tool to do this traversal and such should be its own independent object, because there are many plugins that could make use of this functionality.

Concept worked out via Tasks V-REP plugin issue: https://github.com/jrl-umi3218/Tasks/issues/10
